components/wso2is7.key.manager/src/main/java/org/wso2/is7/client/WSO2IS7KeyManager.java

```diff
@@ -987,26 +1242,87 @@ public void deleteScope(String scopeName) throws APIManagementException {
     @Override
     public void updateScope(Scope scope) throws APIManagementException {
 
-        String scopeKey = scope.getKey();
         try {
-            ScopeDTO scopeDTO = new ScopeDTO();
-            scopeDTO.setDisplayName(scope.getName());
-            scopeDTO.setDescription(scope.getDescription());
-            if (StringUtils.isNotBlank(scope.getRoles()) && scope.getRoles().trim().split(",").length > 0) {
-                scopeDTO.setBindings(Arrays.asList(scope.getRoles().trim().split(",")));
-            }
-            try (Response response = scopeClient.updateScope(scopeDTO, scope.getKey())) {
-                if (response.status() != HttpStatus.SC_OK) {
-                    String responseString = readHttpResponseAsString(response.body());
-                    String errorMessage =
-                            "Error occurred while updating scope: " + scope.getName() + ". Error Status: " +
-                                    response.status() + " . Error Response: " + responseString;
-                    throw new APIManagementException(errorMessage);
+            String wso2IS7APIResourceId = getWSO2IS7APIResourceId();
+            if (wso2IS7APIResourceId != null) {
+                JsonObject payload = new JsonObject();
+                payload.addProperty("displayName", scope.getName());
+                payload.addProperty("description", scope.getDescription());
```

Comment:
TODO: Check if we can use DTO objects instead of JSON payloads.

Reply:
Done via https://github.com/wso2-extensions/apim-km-wso2is/pull/147/commits/ca7b75946e90aebf51621c3e816bf0e117338c60